NuRadioReco/modules/measured_noise/RNO_G/noiseImporter.py

```diff
@@ -0,0 +1,176 @@
+from NuRadioReco.utilities import units
+
+import logging
+
+
+class noiseImporter:
+    """
+    Imports recorded traces from RNOG stations. 
+    
+    """
+
+
+    def begin(self, noise_folders, file_pattern="*",
+              match_station_id=False, station_ids=None,
+              channel_mapping=None, scramble_noise_file_order=True,
+              log_level=logging.INFO, reader_kwargs={}):
+        """
+        
+        Parameters
+        ----------
+        noise_folders: str or list(str)
+            Folder(s) containing noise file(s). Search in any subfolder as well.
+            
+        file_patters: str
+            File patters used to search for directories, (Default: "*", other examples might be "combined")
+            
+        match_station_id: bool
+            If True, add only noise from stations with the same id. (Default: False)
+        
+        station_ids: list(int)
+            Only add noise from those station ids. If None, use any station. (Default: None)
+        
+        channel_mapping: dict or None
+            option relevant for MC studies of new station designs where we do not
+            have forced triggers for. The channel_mapping dictionary maps the channel
+            ids of the MC station to the channel ids of the noise data
+            Default is None which is 1-to-1 mapping
+            
+        scramble_noise_file_order: bool
+            If True, randomize the order of noise files before reading them. (Default: True)
+        
+        log_level: loggging log level
+            the log level, default logging.INFO
+
+        reader_kwargs: dict
+            Optional arguements passed to readRNOGDataMattak
+        """
+        
+        self.logger = logging.getLogger('NuRadioReco.RNOG.noiseImporter')
+        self.logger.setLevel(log_level)
+        
+        self._match_station_id = match_station_id
+        self.__station_ids = station_ids
+        
+        self.__channel_mapping = channel_mapping
+        
+        self.logger.info(f"\n\tMatch station id: {match_station_id}"
+                    f"\n\tUse noise from only those stations: {station_ids}"
+                    f"\n\tUse the following channel mapping: {channel_mapping}"
+                    f"\n\tRandomize sequence of noise files: {scramble_noise_file_order}")
+        
+        if not isinstance(noise_folders, list):
+            noise_folders = [noise_folders]
+        
+        # find all subfolders
+        noise_files = []
+        for noise_folder in noise_folders:
+            noise_files += glob.glob(f"{noise_folder}/**/{file_pattern}root", recursive=True)
+        self.__noise_folders = np.unique([os.path.dirname(e) for e in noise_files])       
+        
+        self.logger.info(f"Found {len(self.__noise_folders)}")
+        if not len(self.__noise_folders):
+            raise ValueError
+                
+        if scramble_noise_file_order:
+            random.shuffle(self.__noise_folders)
```

Comment:
We generally try to use seeded random generation (using either `np.random.RandomState` or `np.random.Philox`) to be able to make things reproducible. I guess alternatively (or maybe additionally) one could store which noise trace has been added to which channel. I don't know how important you think this is, it can also be postponed to a future PR so we can merge the new reader sooner.

Reply:
Implemented the generator. Book-keeping of the random seed and the "noise-trace add to which event" I will label "nice-to-have" for a future PR